src/main/java/picard/fingerprint/FingerprintChecker.java

```diff
@@ -575,6 +582,12 @@ public IntervalList getLociToGenotype(final Collection<Fingerprint> fingerprints
             }
         }
 
+        if (!foundALocus && in.getFileHeader().getSortOrder()!= SAMFileHeader.SortOrder.coordinate) {
+            final PicardException noReadsFound = new PicardException(String.format("Couldn't even find one locus with reads to fingerprint in file %s, which in addition isn't coordinate-sorted. Please sort the file and try again.", samFile));
+            log.error(noReadsFound);
```

Comment:
Is it usual to log before throw in picard? Normally I would expect code to throw and then rely on the catch point to log. This is also usually shorter to write as code has (potentially) many throw points but few catch points.

Reply:
I used it because I wanted to see that the error that is thrown is what I expect...there was a bug previously and this test succeeded by failing in a different way....should change back.